src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -521,10 +1051,13 @@ public void Abort_BeginGetRequestStreamThenAbort_EndGetRequestStreamThrowsWebExc
 
             request.BeginGetResponse(new AsyncCallback(RequestStreamCallback), state);
 
-            request.Abort();
-            Assert.Equal(1, state.RequestStreamCallbackCallCount);
-            WebException wex = state.SavedRequestStreamException as WebException;
-            Assert.Equal(WebExceptionStatus.RequestCanceled, wex.Status);
+            if (!request.HaveResponse)
```

Comment:
Ya there are race conditions and possible deadlocks. This is by design according to MSDN. There is an issue for these tests here.